packages/@react-stately/grid/src/GridCollection.ts

```diff
@@ -79,20 +79,6 @@ export class GridCollection<T> implements IGridCollection<T> {
 
       if (last) {
         last.nextKey = null;
-
-        if (rowHasCellWithColSpan && node.type === 'item') {
```

Comment:
moved this check back to Table's domain
it was counting something about expandable rows as an extra column in the dynamic rendering, though I'm not entirely sure why